### PR TITLE
[ODS-4463] Rename Swagger objects to OpenApiMetadata objects

### PR DESCRIPTION
**Pull request by BhanuprakashThunga**

### CODE REVIEW

File: Application/EdFi.Ods.Features/Controllers/OpenApiMetadataController.cs [hunk 1]
```diff
@@ -115,7 +115,7 @@ string GetMetadataUrlSegmentForCacheItem(OpenApiContent apiContent, int? schoolY
                     ? string.Empty
                     : apiContent.RelativeSectionPath.EnsureSuffixApplied("/");
 
-                return $"{basePath}/{relativeSectionUri}{SwaggerDocumentHelper.SwaggerJson}".ToLowerInvariant();
+                return $"{basePath}/{relativeSectionUri}{OpenApiMetadataDocumentHelper.SwaggerJson}".ToLowerInvariant();
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/Controllers/OpenApiMetadataController.cs:118]:
```suggestion
                return $"{basePath}/{relativeSectionUri}{OpenApiMetadataDocumentHelper.Json}".ToLowerInvariant();
```

Comment (jamessmckay) [reply Application/EdFi.Ods.Features/Controllers/OpenApiMetadataController.cs:118]: make sure you rename using resharper to get all the changes

File: Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataDocumentHelper.cs [hunk 2]
```diff
@@ -164,28 +166,30 @@ public static Schema CreateReferenceSchema(Reference reference)
                 });
 
             return new Schema
-                   {
-                       type = "object", required = properties.Where(x => x.IsRequired)
+            {
+                type = "object",
+                required = properties.Where(x => x.IsRequired)
                                                              .Select(x => x.PropertyName.ToCamelCase())
                                                              .ToList(),
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataDocumentHelper.cs:173]: fix formatting please

File: Application/EdFi.Ods.Features/OpenApiMetadata/Providers/OpenApiMetadataCacheProvider.net.cs [hunk 3]
```diff
@@ -153,8 +153,8 @@ IEnumerable<OpenApiContent> CreateSwaggerUiSection()
                             OpenApiMetadataSections.SwaggerUi,
                             x.Key, new Lazy<string>(
                                 () =>
-                                    new SwaggerDocumentFactory(
-                                        new SwaggerDocumentContext(_resourceModelProvider.GetResourceModel())).Create(x.Value)),
+                                    new OpenApiMetadataDocumentFactory(
+                                        new OpenApiMetadataDocumentContext(_resourceModelProvider.GetResourceModel())).Create(x.Value)),
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Providers/OpenApiMetadataCacheProvider.net.cs:157]: lets reformat this document correctly please...

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy.cs [hunk 4]
```diff
@@ -11,44 +11,45 @@
 
 namespace EdFi.Ods.Features.OpenApiMetadata.Strategies.FactoryStrategies
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy.cs:12]: lets reformat this whole document please. its a little wonky

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiCompositeStrategy.cs [hunk 5]
```diff
@@ -55,15 +55,17 @@ public IEnumerable<SwaggerResource> GetFilteredResources(SwaggerDocumentContext
...
+                        Readable = true,
+                        CompositeResourceContext = new CompositeResourceContext
+                        {
+                            OrganizationCode = swaggerDocumentContext.CompositeContext.OrganizationCode,
+                            Specification = compositeDefinition.Element("Specification"),
+                            BaseResource =
                                                                             compositeDefinition.Element("BaseResource")
                                                                                                .AttributeValue("name")
-                                                                    }
+                        }
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiCompositeStrategy.cs:68]: fix formatting please

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiMetadataUiDescriptorOnlyStrategy.cs [hunk 6]
```diff
@@ -10,15 +10,15 @@
 
 namespace EdFi.Ods.Features.OpenApiMetadata.Strategies.ResourceStrategies
 {
-    public class SwaggerUiDescriptorOnlyStrategy : IOpenApiMetadataResourceStrategy
+    public class OpenApiMetadataUiDescriptorOnlyStrategy : IOpenApiMetadataResourceStrategy
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiMetadataUiDescriptorOnlyStrategy.cs:13]: lets reformat this whole document please

File: Application/EdFi.Ods.Features/OpenApiMetadata/Dtos/OpenApiMetadataDocumentContext.cs [hunk 7]
```diff
@@ -37,19 +37,19 @@ public SwaggerDocumentContext(IResourceModel resourceModel)
...
         AllWithStronglyTypedExtensions, /// <summary>
-        /// Uses generalized extension collections (for a loosely-coupled client-side SDK).
-        /// </summary>
+                                        /// Uses generalized extension collections (for a loosely-coupled client-side SDK).
+                                        /// </summary>
         GeneralizedExtensions, /// <summary>
-        /// Includes extension resources only and any model definitions for "entity extension" classes.
-        /// </summary>
+                               /// Includes extension resources only and any model definitions for "entity extension" classes.
+                               /// </summary>
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Dtos/OpenApiMetadataDocumentContext.cs:52]: fix this formatting please and revert the white space changes

File: Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataTagsFactory.cs [hunk 8]
```diff
@@ -12,25 +12,25 @@
...
         {
-            return swaggerResources.Where(x => _filterStrategy.ShouldInclude(x.Resource))
+            return openApiMetadataResources.Where(x => _filterStrategy.ShouldInclude(x.Resource))
                                    .Select(
                                         x => new Tag
-                                             {
-                                                 name = SwaggerDocumentHelper.GetResourcePluralName(x.Resource)
+                                        {
+                                            name = OpenApiMetadataDocumentHelper.GetResourcePluralName(x.Resource)
                                                                              .ToCamelCase(),
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataTagsFactory.cs:31]: this formatting needs to be fixed please

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataCompositePathsFactoryStrategy.cs [hunk 9]
```diff
@@ -17,107 +17,111 @@
...
                                                                         r.CompositeResourceContext,
-                                                                    Readable = true, Path =
-                                                                        $"/{_swaggerDocumentContext.CompositeContext.CategoryName}/{r.Resource.PluralName}",
-                                                                    RequestProperties = GetMetaDataPropertiesForCurrentResource(r)
-                                                                })
+                                                               Readable = true,
+                                                               Path =
+                                                                        $"/{_openApiMetadataDocumentContext.CompositeContext.CategoryName}/{r.Resource.PluralName}",
+                                                               RequestProperties = GetMetaDataPropertiesForCurrentResource(r)
+                                                           })
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataCompositePathsFactoryStrategy.cs:42]: this whole section needs to be reformatted correctly, also there is extra blank lines in it.

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataCompositePathsFactoryStrategy.cs [hunk 10]
```diff
@@ -17,107 +17,111 @@
...
                                           .Where(
                                                route =>
                                                    !IsDefaultRoute(route)
-                                                   && RouteAppliesToResource(route, swaggerResource))
+                                                   && RouteAppliesToResource(route, openApiMetadataResource))
                                           .SelectMany(
                                                d =>
                                                {
-                                                   var operationSpecSuffix = GetOperationSpecNickname(swaggerResource.Resource, d);
+                                                   var operationSpecSuffix = GetOperationSpecNickname(openApiMetadataResource.Resource, d);
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataCompositePathsFactoryStrategy.cs:65]: lets fix the formatting on this method please

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy.cs [hunk 11]
```diff
@@ -11,44 +11,45 @@
...
 {
-    public class SwaggerDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy : ISwaggerDefinitionsFactoryEdFiExtensionBridgeStrategy
+    public class OpenApiMetadataDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy : IOpenApiMetadataDefinitionsFactoryEdFiExtensionBridgeStrategy
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy.cs:14]: The comment for this was not completed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiMetadataUiDescriptorOnlyStrategy.cs (hunk 6 above)
Comment (jamessmckay) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiMetadataUiDescriptorOnlyStrategy.cs:13]: this comment was not completed

File: Application/EdFi.Ods.Features/Controllers/OpenApiMetadataController.cs (hunk 1 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/Controllers/OpenApiMetadataController.cs]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataDocumentHelper.cs (hunk 2 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataDocumentHelper.cs]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Providers/OpenApiMetadataCacheProvider.net.cs (hunk 3 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Providers/OpenApiMetadataCacheProvider.net.cs]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy.cs (hunk 4 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy.cs:12]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiCompositeStrategy.cs (hunk 5 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiCompositeStrategy.cs]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiMetadataUiDescriptorOnlyStrategy.cs (hunk 6 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/ResourceStrategies/OpenApiMetadataUiDescriptorOnlyStrategy.cs:13]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Dtos/OpenApiMetadataDocumentContext.cs (hunk 7 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Dtos/OpenApiMetadataDocumentContext.cs]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataTagsFactory.cs (hunk 8 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataTagsFactory.cs]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataCompositePathsFactoryStrategy.cs (hunk 9 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataCompositePathsFactoryStrategy.cs]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataCompositePathsFactoryStrategy.cs (hunk 10 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataCompositePathsFactoryStrategy.cs]: Fixed

File: Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy.cs (hunk 11 above)
Comment (syednoorulla) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Strategies/FactoryStrategies/OpenApiMetadataDefinitionsFactoryDefaultEdFiExtensionBridgeStrategy.cs:14]: Fixed

**Pull request merged**